File: java-solutions/Qn8.java
public class Qn8
{
    public static int validNepName(String name)
    {
        name = name.trim();
        String[] arr = name.split(" ");
        if(arr.length <= 1) return-1;

        int spaces = 0;
        for(String s: arr)
        {
            if(!s.equals(" ") && !s.equals(""))
                spaces++;
        }
        return spaces;
    }

    public static void main(String[] args)
    {
        String name1 = "Ramesh Poudel";
        String name2 = " Ramesh p Poudel";
        String name3 = "RameshPoudel";
        String name4 = "Ramesh Poudel ";
        System.out.println(validNepName(name1));
        System.out.println(validNepName(name2));
        System.out.println(validNepName(name3));
        System.out.println(validNepName(name4));
    }
}

File: rust-solutions/qn8.rs
fn valid_nep_name(name: &mut String) -> i32
{
    let mut spaces = 0;
    let ant = name.trim();
    for c in ant.chars()
    {
        if c == ' '
        {
            spaces += 1;
        }
    }
    if spaces == 0
    {
        return -1;
    }
    return spaces + 1;
}

fn main()
{
    let mut name = String::from(" Ramesh");
    println!("{}", valid_nep_name(&mut name));


    let mut name2 = String::from(" Ramesh Poudel");
    println!("{}", valid_nep_name(&mut name2));

    
    let mut name3 = String::from(" Ramesh Prasad Poudel  ");
    println!("{}", valid_nep_name(&mut name3));
}
